## 76% of end-end tests have passed
```
Total executed tests: 51
Failed tests: 10
Expected to fail tests: 2
Passed tests: 39
```
```
IDs of failed tests: 727230,702809,702733,704613,702948,727229,702786,702845,702775,703629 
```
```
IDs of expected to fail tests: 703495,727232 
```
<h3>Failed tests (10)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=ilmotta/harden-bignumber-comparator&PR_ID=20614&APK_NAME=20614.apk&TR_CASE_IDS=727230,702809,702733,704613,702948,727229,702786,702845,702775,703629">Rerun failed tests</a></li><br/><h4>Class TestDeepLinksOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593257">test_links_open_universal_links_from_chat</a>, id: 704613 </b></td></tr><tr><td><p><blockquote><div>Device 1: Looking for a message by text: https://status.app/c/GzAAAORtwyW4xNWM4td0F7hOnYZ1apSqCCRUUR0qxD19n3Ec97fX_aIVIGFWbdUM#zQ3shk6dgK8dYWWSC4m8Jj5c91zyfhfj1fFkgypS8D9gsXkrK </div><div>Device 1: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'https://status.app/c/GzAAAORtwyW4xNWM4td0F7hOnYZ1apSqCCRUUR0qxD19n3Ec97fX_aIVIGFWbdUM#zQ3shk6dgK8dYWWSC4m8Jj5c91zyfhfj1fFkgypS8D9gsXkrK')]` </div></blockquote></p>

```
critical/test_deep_and_universal_links.py:71: in test_links_open_universal_links_from_chat
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Community 'Open community for e2e' was not requested to join by the url https://status.app/c/GzwAAGS9TbI9CiYaoZj0TRNelWAPwk8gSHIEmqgqb0ADPw5rnFvk6bp_n3BlZxVMshmxFyI=#zQ3shkrmSmwb66sw2EuUvC5iqiNqEeevTfjtks3p8ZaaCYoNj
E    Community 'Closed community' was not requested to join by the url https://status.app/c/GzAAAORtwyW4xNWM4td0F7hOnYZ1apSqCCRUUR0qxD19n3Ec97fX_aIVIGFWbdUM#zQ3shk6dgK8dYWWSC4m8Jj5c91zyfhfj1fFkgypS8D9gsXkrK
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5180400f58e142018a553cb4c7625996?auth=483fe364255423a4afd62e485fafde29#52">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593218">test_links_deep_links</a>, id: 702775 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `BrowserTab` by `accessibility id`: `browser-stack-tab` </div><div>Device 1: Tap on found: BrowserTab </div></blockquote></p>

```
critical/test_deep_and_universal_links.py:115: in test_links_deep_links
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Community 'Open community for e2e' was not requested to join by the deep link status.app://c/GzwAAGS9TbI9CiYaoZj0TRNelWAPwk8gSHIEmqgqb0ADPw5rnFvk6bp_n3BlZxVMshmxFyI=#zQ3shkrmSmwb66sw2EuUvC5iqiNqEeevTfjtks3p8ZaaCYoNj
E    Community 'Closed community' was not requested to join by the deep link status.app://c/GzAAAORtwyW4xNWM4td0F7hOnYZ1apSqCCRUUR0qxD19n3Ec97fX_aIVIGFWbdUM#zQ3shk6dgK8dYWWSC4m8Jj5c91zyfhfj1fFkgypS8D9gsXkrK
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5180400f58e142018a553cb4c7625996?auth=483fe364255423a4afd62e485fafde29#201">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593215">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 1: Getting PN by 'DELETE ME' </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:531: in test_1_1_chat_text_message_delete_push_disappear
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Message status was not changed to 'Delivered' after 60 s
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c7affce9ddcc4befae4aff458cb1c10c?auth=2c64802a48e7579a058ef34d44644760#552">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/844c40bd9ebc48998b39194cea3c8719?auth=8d7a3b3329906dfe9c75e40b8b4f7ec3#410">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593260">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td><p><blockquote><div># STEP: Getting ETH amount in the wallet of the sender before transaction</div><div>Device 1: Find `WalletTab` by `accessibility id`: `wallet-stack-tab` </div></blockquote></p>

```
critical/test_wallet.py:152: in test_wallet_send_asset_from_drawer
    sender_balance, receiver_balance, eth_amount_sender, eth_amount_receiver = self._get_balances_before_tx()
critical/test_wallet.py:40: in _get_balances_before_tx
    self.wallet_1.wallet_tab.click()
../views/base_element.py:90: in click
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: WalletTab by accessibility id: `wallet-stack-tab` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593259">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `top-bar` </div><div>Device 2: Find `Button` by `accessibility id`: `top-bar` </div></blockquote></p>

```
critical/test_wallet.py:147: in test_wallet_send_eth
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Sender balance is not updated on Etherscan, it is 0.0011317461 but expected to be 0.0011338461
```

<br/><br/></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593226">test_community_markdown_support</a>, id: 702809 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:867: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:419: in join_community
    ChatView(self.driver).chat_element_by_text("https://status.app/c/").click_on_link_inside_message_body()
../views/chat_view.py:175: in click_on_link_inside_message_body
    self.message_body.wait_for_visibility_of_element(30)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Text by xpath:`//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'https://status.app/c/')]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593243">test_community_hashtag_links_to_community_channels</a>, id: 702948 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:867: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:419: in join_community
    ChatView(self.driver).chat_element_by_text("https://status.app/c/").click_on_link_inside_message_body()
../views/chat_view.py:175: in click_on_link_inside_message_body
    self.message_body.wait_for_visibility_of_element(30)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Text by xpath:`//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'https://status.app/c/')]` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[Can't enter channel after community is fetched for the first time, https://github.com/status-im/status-mobile/issues/20395]]</b><br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593223">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><p><blockquote><div>Device 2: Joining community </div><div>Device 2: Looking for a message by text: https://status.app/c/ </div></blockquote></p>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:867: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:419: in join_community
    ChatView(self.driver).chat_element_by_text("https://status.app/c/").click_on_link_inside_message_body()
../views/chat_view.py:175: in click_on_link_inside_message_body
    self.message_body.wait_for_visibility_of_element(30)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Text by xpath:`//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'https://status.app/c/')]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d61158e47a8e4f39ad39885c98797dbc?auth=445a5706671c521ce12719cc48f2024b">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/83cf5ec7d88d429a94a59180111f76d5?auth=ef51b97af116385cebf52ed2f6632409">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593234">test_community_leave</a>, id: 702845 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:867: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:419: in join_community
    ChatView(self.driver).chat_element_by_text("https://status.app/c/").click_on_link_inside_message_body()
../views/chat_view.py:175: in click_on_link_inside_message_body
    self.message_body.wait_for_visibility_of_element(30)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Text by xpath:`//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'https://status.app/c/')]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593256">test_community_join_when_node_owner_offline</a>, id: 703629 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:867: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:419: in join_community
    ChatView(self.driver).chat_element_by_text("https://status.app/c/").click_on_link_inside_message_body()
../views/chat_view.py:175: in click_on_link_inside_message_body
    self.message_body.wait_for_visibility_of_element(30)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Text by xpath:`//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'https://status.app/c/')]` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[Can't enter channel after community is fetched for the first time, https://github.com/status-im/status-mobile/issues/20395]]</b><br/><br/></td></tr></tbody></table></details><h3>Expected to fail tests (2)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593253">test_group_chat_mute_chat</a>, id: 703495 </b></td></tr><tr><td><p><blockquote><div># STEP: Change device time so chat will be unmuted by timer</div><div>Device 2: Long press on `ChatElement` </div></blockquote></p>

```
critical/chats/test_group_chat.py:497: in test_group_chat_mute_chat
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Chat is still muted after timeout 
```

<b>[[Chat is not unmuted after expected time: https://github.com/status-im/status-mobile/issues/19627]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/27d4cda794f643fbb5ee7cf3dbc1df06?auth=0d2539933a00b1d49a29634d06d99473#481">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2ecffce982e04d4ea6eece2da95e9865?auth=d7693c2ab21fbb6473aa809bd20f2029#273">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9b08e20a7d1c4558ae28fbca7add8bbb?auth=3ccb634e96bc7ec59c026abab03d651f#242">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593262">test_wallet_add_remove_watch_only_account</a>, id: 727232 </b></td></tr><tr><td><p><blockquote><div>Device 1: `Text` is `0x8d2413447ff297d30bdc475f6d5cb00254685aae` </div><div>Device 1: Click system back button </div></blockquote></p>

```
critical/test_wallet.py:256: in test_wallet_add_remove_watch_only_account
    self.home_view.driver.fail(
base_test_case.py:178: in fail
    pytest.fail('Device %s: %s' % (self.number, text))
 Device 1: Incorrect address '0x8d2413447ff297d30bdc475f6d5cb00254685aae' is shown when swiping between accounts, expected one is '0x8d2413447ff297d30bdc475f6d5cb00254685aae' 
```

<b>[[Missing networks in account address, https://github.com/status-im/status-mobile/issues/20166]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ac034ddc7a0a4e0a858c042e853807fe?auth=0b171115049eff8c7a8289ea1af3949e#102">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (39)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593248">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cc00c448d664644886700d7bf306358?auth=a7be9ce9a8520efd51b632c7016bc983#233">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9598dbc2deda4832bb37a385899f0d5d?auth=384cb3992dbd59a435e6ba328b4b8da5#165">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593239">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cc00c448d664644886700d7bf306358?auth=a7be9ce9a8520efd51b632c7016bc983#258">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9598dbc2deda4832bb37a385899f0d5d?auth=384cb3992dbd59a435e6ba328b4b8da5#226">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593230">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cc00c448d664644886700d7bf306358?auth=a7be9ce9a8520efd51b632c7016bc983#306">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9598dbc2deda4832bb37a385899f0d5d?auth=384cb3992dbd59a435e6ba328b4b8da5#286">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593246">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cc00c448d664644886700d7bf306358?auth=a7be9ce9a8520efd51b632c7016bc983#552">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9598dbc2deda4832bb37a385899f0d5d?auth=384cb3992dbd59a435e6ba328b4b8da5#470">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593241">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cc00c448d664644886700d7bf306358?auth=a7be9ce9a8520efd51b632c7016bc983#425">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9598dbc2deda4832bb37a385899f0d5d?auth=384cb3992dbd59a435e6ba328b4b8da5#394">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593258">test_community_edit_delete_message_when_offline</a>, id: 704615 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cc00c448d664644886700d7bf306358?auth=a7be9ce9a8520efd51b632c7016bc983#587">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9598dbc2deda4832bb37a385899f0d5d?auth=384cb3992dbd59a435e6ba328b4b8da5#496">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593229">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cc00c448d664644886700d7bf306358?auth=a7be9ce9a8520efd51b632c7016bc983#200">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9598dbc2deda4832bb37a385899f0d5d?auth=384cb3992dbd59a435e6ba328b4b8da5#136">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593228">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cc00c448d664644886700d7bf306358?auth=a7be9ce9a8520efd51b632c7016bc983#158">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9598dbc2deda4832bb37a385899f0d5d?auth=384cb3992dbd59a435e6ba328b4b8da5#107">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593233">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cc00c448d664644886700d7bf306358?auth=a7be9ce9a8520efd51b632c7016bc983#330">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9598dbc2deda4832bb37a385899f0d5d?auth=384cb3992dbd59a435e6ba328b4b8da5#303">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593232">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cc00c448d664644886700d7bf306358?auth=a7be9ce9a8520efd51b632c7016bc983#176">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9598dbc2deda4832bb37a385899f0d5d?auth=384cb3992dbd59a435e6ba328b4b8da5#124">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>11. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593231">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4cc00c448d664644886700d7bf306358?auth=a7be9ce9a8520efd51b632c7016bc983#395">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9598dbc2deda4832bb37a385899f0d5d?auth=384cb3992dbd59a435e6ba328b4b8da5#364">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593219">test_add_contact_field_validation</a>, id: 702777 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a008bbac8f6e49fc8579493ca44e3061?auth=c8ed6533cb9a3487b00febcd9eb9846b#134">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2beb3238ce694240af665a3095a17cbd?auth=6e3c0ab4386dcea3f4c6a9170ee8feea#218">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593237">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a008bbac8f6e49fc8579493ca44e3061?auth=c8ed6533cb9a3487b00febcd9eb9846b#99">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2beb3238ce694240af665a3095a17cbd?auth=6e3c0ab4386dcea3f4c6a9170ee8feea#121">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593236">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a008bbac8f6e49fc8579493ca44e3061?auth=c8ed6533cb9a3487b00febcd9eb9846b#56">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2beb3238ce694240af665a3095a17cbd?auth=6e3c0ab4386dcea3f4c6a9170ee8feea#55">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593247">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2190c52152fc4ad1b802b6917c458176?auth=6e9da969de31dfbe9d8a2edc4a8713e4#229">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593216">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2190c52152fc4ad1b802b6917c458176?auth=6e9da969de31dfbe9d8a2edc4a8713e4#76">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593255">test_community_discovery</a>, id: 703503 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2190c52152fc4ad1b802b6917c458176?auth=6e9da969de31dfbe9d8a2edc4a8713e4#354">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593240">test_community_undo_delete_message</a>, id: 702869 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2190c52152fc4ad1b802b6917c458176?auth=6e9da969de31dfbe9d8a2edc4a8713e4#105">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593235">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2190c52152fc4ad1b802b6917c458176?auth=6e9da969de31dfbe9d8a2edc4a8713e4#52">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593251">test_community_mute_community_and_channel</a>, id: 703382 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2190c52152fc4ad1b802b6917c458176?auth=6e9da969de31dfbe9d8a2edc4a8713e4#124">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593220">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c7affce9ddcc4befae4aff458cb1c10c?auth=2c64802a48e7579a058ef34d44644760#149">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/844c40bd9ebc48998b39194cea3c8719?auth=8d7a3b3329906dfe9c75e40b8b4f7ec3#121">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593227">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c7affce9ddcc4befae4aff458cb1c10c?auth=2c64802a48e7579a058ef34d44644760#411">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/844c40bd9ebc48998b39194cea3c8719?auth=8d7a3b3329906dfe9c75e40b8b4f7ec3#261">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593217">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c7affce9ddcc4befae4aff458cb1c10c?auth=2c64802a48e7579a058ef34d44644760#353">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/844c40bd9ebc48998b39194cea3c8719?auth=8d7a3b3329906dfe9c75e40b8b4f7ec3#211">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593238">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c7affce9ddcc4befae4aff458cb1c10c?auth=2c64802a48e7579a058ef34d44644760#460">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/844c40bd9ebc48998b39194cea3c8719?auth=8d7a3b3329906dfe9c75e40b8b4f7ec3#292">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593252">test_1_1_chat_send_image_save_and_share</a>, id: 703391 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c7affce9ddcc4befae4aff458cb1c10c?auth=2c64802a48e7579a058ef34d44644760#491">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/844c40bd9ebc48998b39194cea3c8719?auth=8d7a3b3329906dfe9c75e40b8b4f7ec3#350">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593213">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c7affce9ddcc4befae4aff458cb1c10c?auth=2c64802a48e7579a058ef34d44644760#211">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/844c40bd9ebc48998b39194cea3c8719?auth=8d7a3b3329906dfe9c75e40b8b4f7ec3#166">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593212">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c7affce9ddcc4befae4aff458cb1c10c?auth=2c64802a48e7579a058ef34d44644760#95">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/844c40bd9ebc48998b39194cea3c8719?auth=8d7a3b3329906dfe9c75e40b8b4f7ec3#84">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593222">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1711b30d41234aa1b12adbea41772ae6?auth=6b6d37b08008b16de3d43e9014e05d7d#171">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e78190febb8046bca78b3c605ca064f9?auth=4d3cec106957fc46278b2f0bdb48fcab#129">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593221">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1711b30d41234aa1b12adbea41772ae6?auth=6b6d37b08008b16de3d43e9014e05d7d#95">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e78190febb8046bca78b3c605ca064f9?auth=4d3cec106957fc46278b2f0bdb48fcab#84">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593254">test_1_1_chat_mute_chat</a>, id: 703496 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1711b30d41234aa1b12adbea41772ae6?auth=6b6d37b08008b16de3d43e9014e05d7d#105">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/e78190febb8046bca78b3c605ca064f9?auth=4d3cec106957fc46278b2f0bdb48fcab#116">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593261">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ac034ddc7a0a4e0a858c042e853807fe?auth=0b171115049eff8c7a8289ea1af3949e#38">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593214">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/27d4cda794f643fbb5ee7cf3dbc1df06?auth=0d2539933a00b1d49a29634d06d99473#361">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2ecffce982e04d4ea6eece2da95e9865?auth=d7693c2ab21fbb6473aa809bd20f2029#230">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9b08e20a7d1c4558ae28fbca7add8bbb?auth=3ccb634e96bc7ec59c026abab03d651f#228">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593250">test_group_chat_send_image_save_and_share</a>, id: 703297 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/27d4cda794f643fbb5ee7cf3dbc1df06?auth=0d2539933a00b1d49a29634d06d99473#288">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2ecffce982e04d4ea6eece2da95e9865?auth=d7693c2ab21fbb6473aa809bd20f2029#168">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9b08e20a7d1c4558ae28fbca7add8bbb?auth=3ccb634e96bc7ec59c026abab03d651f#141">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593249">test_group_chat_reactions</a>, id: 703202 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/27d4cda794f643fbb5ee7cf3dbc1df06?auth=0d2539933a00b1d49a29634d06d99473#177">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2ecffce982e04d4ea6eece2da95e9865?auth=d7693c2ab21fbb6473aa809bd20f2029#106">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9b08e20a7d1c4558ae28fbca7add8bbb?auth=3ccb634e96bc7ec59c026abab03d651f#86">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593224">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/27d4cda794f643fbb5ee7cf3dbc1df06?auth=0d2539933a00b1d49a29634d06d99473#160">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2ecffce982e04d4ea6eece2da95e9865?auth=d7693c2ab21fbb6473aa809bd20f2029#78">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9b08e20a7d1c4558ae28fbca7add8bbb?auth=3ccb634e96bc7ec59c026abab03d651f#78">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593225">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/27d4cda794f643fbb5ee7cf3dbc1df06?auth=0d2539933a00b1d49a29634d06d99473#328">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2ecffce982e04d4ea6eece2da95e9865?auth=d7693c2ab21fbb6473aa809bd20f2029#203">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9b08e20a7d1c4558ae28fbca7add8bbb?auth=3ccb634e96bc7ec59c026abab03d651f#201">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePRTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593244">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b70183d909c446ce934b848129ad833a?auth=91313662f64d024305a8aa56dd430fa0#132">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2c4a7209b87b437aab8ab4f69192c5bd?auth=5f00de43bba7d08a69421a588ec9ddec#102">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593245">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b70183d909c446ce934b848129ad833a?auth=91313662f64d024305a8aa56dd430fa0#173">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2c4a7209b87b437aab8ab4f69192c5bd?auth=5f00de43bba7d08a69421a588ec9ddec#118">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3593242">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f6b8c235712f48f5b98c410847194f8c?auth=a95e21a91b5719cb0fd98d35f87d7604#132">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/556899313d9c4ab897af6d705a037eae?auth=1e17501efe96691e1f51ed7648bb7e9c#98">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/utils/money.cljs
+++ b/src/utils/money.cljs
@@ -61,7 +61,11 @@
 (extend-type BigNumber
  IEquiv
    (-equiv [this other]
-     (equal-to this other))
+     (if (or (number? other)
+             (string? other)
+             (instance? BigNumber other))
+       (equal-to this other)
+       false))
 
  IComparable
    (-compare [this other]

--- a/src/utils/money_test.cljs
+++ b/src/utils/money_test.cljs
@@ -4,26 +4,45 @@
     [utils.money :as money]))
 
 (deftest comparable-test
-  (is (= [(money/bignumber -4)
-          (money/bignumber 0)
-          (money/bignumber 1)
-          (money/bignumber 1.1)
-          (money/bignumber 2.1)]
-         (sort [(money/bignumber 0)
-                (money/bignumber 2.1)
-                (money/bignumber -4)
-                (money/bignumber 1.1)
-                (money/bignumber 1)]))))
+  (testing "sorts bignumbers (and nils)"
+    (is (= [nil
+            nil
+            (money/bignumber -4)
+            (money/bignumber 0)
+            (money/bignumber 1)
+            (money/bignumber 1.1)
+            (money/bignumber 2.1)]
+           (sort [(money/bignumber 0)
+                  nil
+                  (money/bignumber 2.1)
+                  (money/bignumber -4)
+                  (money/bignumber 1.1)
+                  (money/bignumber nil)
+                  (money/bignumber 1)]))))
+
+  (testing "throws when comparing non-bignumbers with bignumbers"
+    (is (thrown? js/Error (sort [(money/bignumber 42) ""])))))
 
 (deftest equivalence-test
-  (is (= (money/bignumber 0)
-         (money/bignumber 0)))
-  (is (= (money/bignumber -1)
-         (money/bignumber -1)))
-  (is (not (= (money/bignumber 10)
-              (money/bignumber -10))))
-  (is (match? {:a {:b {:c (money/bignumber 42)}}}
-              {:a {:b {:c (money/bignumber 42)}}})))
+  (testing "equivalence with numbers and strings"
+    (is (= (money/bignumber 42) 42))
+    (is (= (money/bignumber 42) 42.0))
+    (is (= (money/bignumber 42) "42"))
+    (is (= (money/bignumber 42) "42.0"))
+    (is (not= (money/bignumber 42) :42))
+    (is (not= (money/bignumber 42) {:x 42})))
+
+  (testing "bignumbers are never equivalent to nil"
+    ;; The native `.eq` function throws with nil values, but we gracefully handle this.
+    (is (false? (= (money/bignumber 0) nil)))
+    (is (false? (= (money/bignumber 10) (money/bignumber nil)))))
+
+  (testing "equivalence of actual bignumbers"
+    (is (= (money/bignumber 0) (money/bignumber 0)))
+    (is (= (money/bignumber -1) (money/bignumber -1)))
+    (is (not= (money/bignumber 10) (money/bignumber -10)))
+    (is (match? {:a {:b {:c (money/bignumber 42)}}}
+                {:a {:b {:c (money/bignumber 42)}}}))))
 
 (deftest wei->ether-test
   (testing "Numeric input, 15 significant digits"

